src/par_iter/from_par_iter.rs

```diff
@@ -46,8 +78,11 @@ impl<PAR_ITER, K, V, S> FromParallelIterator<PAR_ITER> for HashMap<K, V, S>
           S: BuildHasher + Default + Send,
 {
     fn from_par_iter(par_iter: PAR_ITER) -> Self {
-        let vec: LinkedList<_> = par_iter.collect();
-        vec.into_iter().collect()
+        let list = collect_list_vec(par_iter);
```

Comment:
Nit: it's probably worth adding a comment referencing the benchmarks in `rayon-demo` showing different strategies.